modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/GridDhtPartitionTopologyImpl.java

```diff
@@ -2708,23 +2690,30 @@ private void removeNode(UUID nodeId) {
     @Override public void ownMoving(AffinityTopologyVersion rebFinishedTopVer) {
         lock.writeLock().lock();
 
-        AffinityTopologyVersion lastAffChangeVer = ctx.exchange().lastAffinityChangedTopologyVersion(lastTopChangeVer);
+        try {
+            AffinityTopologyVersion lastAffChangeVer = ctx.exchange().lastAffinityChangedTopologyVersion(lastTopChangeVer);
 
-        if (lastAffChangeVer.compareTo(rebFinishedTopVer) > 0)
-            log.info("Affinity topology changed, no MOVING partitions will be owned " +
-                "[rebFinishedTopVer=" + rebFinishedTopVer +
-                ", lastAffChangeVer=" + lastAffChangeVer + "]");
+            if (lastAffChangeVer.compareTo(rebFinishedTopVer) > 0) {
+                if (log.isInfoEnabled()) {
+                    log.info("Affinity topology changed, no MOVING partitions will be owned " +
+                        "[rebFinishedTopVer=" + rebFinishedTopVer +
+                        ", lastAffChangeVer=" + lastAffChangeVer + "]");
+                }
 
-        try {
-            for (GridDhtLocalPartition locPart : grp.topology().currentLocalPartitions()) {
+                grp.preloader().forceRebalance();
```

Comment:
This is poor solution because it breaks rebalancing compatibility optimization.
We should always own partitions if topology changes are compatible.
Also we need explicit test for this scenario.

Reply:
I suggest that will be a temporary solution.
Until we haven't implement the issue
https://ggsystems.atlassian.net/browse/GG-27642